test/e2e_federation/util.go

```diff
@@ -510,3 +510,52 @@ func waitForReplicaSetToBeDeletedOrFail(clientset *fedclientset.Clientset, names
 		framework.Failf("Error in deleting replica set %s: %v", replicaSet, err)
 	}
 }
+
+// waitForDeploymentToBeDeletedOrFail waits for the named deployment in namespace to be deleted.
+// If the deletion fails, the enclosing test fails.
+func waitForDeploymentToBeDeletedOrFail(clientset *fedclientset.Clientset, namespace string, deployment string) {
+	err := wait.Poll(5*time.Second, fedframework.FederatedDefaultTestTimeout, func() (bool, error) {
```

Comment:
Please don't embed constants in code like this.  There are e2e-wide constants for these delays.  Please use them.